src/screens/ProfileScreen.tsx

```diff
@@ -1,11 +1,166 @@
-import { Box, Text } from "native-base";
+import * as WebBrowser from "expo-web-browser";
+import {
+  Box,
+  Button,
+  Text,
+  Heading,
+  FormControl,
+  Input,
+  Radio,
+  VStack,
+  ScrollView,
+  useToast,
+} from "native-base";
 import { FC } from "react";
+import { Resolver, Controller, useForm } from "react-hook-form";
+
+import { updateUserProfile, logOut } from "../actions/user.actions";
+import { useAppDispatch, useAppSelector } from "../hooks/redux";
+
+type FormValues = {
+  name: string;
+  avatar: "LUP" | "VALU";
+};
+
+const resolver: Resolver<FormValues> = async (values) => {
+  return {
+    values: values.name ? values : {},
+    errors: !values.name
+      ? {
+          name: {
+            type: "required",
+            message: "El nombre es requerido",
+          },
+        }
+      : {},
+  };
+};
 
 const ProfileScreen: FC = () => {
+  const _handlePressButtonAsync = async () => {
+    await WebBrowser.openBrowserAsync("https://www.yadirapg.com/");
+  };
+
+  const dispatch = useAppDispatch();
+  const toast = useToast();
+
+  const { profile, error } = useAppSelector((state) => state.user);
+
+  const {
+    control,
+    handleSubmit,
+    watch,
+    formState: { errors },
+  } = useForm<FormValues>({
+    resolver,
+    defaultValues: {
+      avatar: profile?.avatar,
+      name: profile?.name,
+    },
+  });
+
+  const avatar = watch("avatar");
+
+  const onSubmit = async (data: FormValues) => {
+    if (profile!.avatar == data.avatar && profile!.name == data.name) return;
+    dispatch(updateUserProfile({ avatar: data.avatar, name: data.name, id: profile!.id }));
+    toast.show({
+      title: "Información actualizada",
+      description: "Tu información ha sido actualizada exitosamente",
+      status: "success",
+    });
+    if (error) {
```

Comment:
If an error occurs, both toasts will be shown. Try using an else if statement, or an early return to prevent this behaviour 😄

Reply:
changed! :)